ompi/mpi/c/aint_add_abi.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2004-2007 The Trustees of Indiana University and Indiana
+ *                         University Research and Technology
+ *                         Corporation.  All rights reserved.
+ * Copyright (c) 2004-2005 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+#include "ompi_config.h"
+#include <stdio.h>
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/mpi/c/abi.h"
+#include "ompi/mpi/c/abi_converters.h"
+
+#if OMPI_BUILD_MPI_PROFILING
+#if OPAL_HAVE_WEAK_SYMBOLS
+#pragma weak MPI_Aint_add = PMPI_Aint_add
+#endif
+#define MPI_Aint_add PMPI_Aint_add
+#endif
+
+MPI_Aint MPI_Aint_add(MPI_Aint addr1, MPI_Aint addr2)
+{
+    return (addr1 + addr2);
```

Comment:
```suggestion
    return (MPI_Aint) ((char *) (base) + (disp));
```

Reply:
I know I'm being pedantic, but let me quote the MPI standard (5.0 pp 146)

> To ensure portability, arithmetic on absolute addresses should not be
performed with the intrinsic operators “-” and “+”.